lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -1176,9 +1176,13 @@ synchronized List<PresenceMessage> endSync() {
                 /* any members that were present at the start of the sync,
                  * and have not been seen in sync, can be removed */
                 for(String itemKey: residualMembers) {
-                    /* clone presence message as it still can be in the internal presence map */
-                    removedEntries.add((PresenceMessage)members.get(itemKey).clone());
-                    members.remove(itemKey);
+                    PresenceMessage removedMember = members.remove(itemKey);
```

Comment:
A concurrency issue was my first guess, but I've ruled it out based on the fact that all modifications of (and most of the interactions with) `residualMembers` and `members` are inside `synchronized` methods. Do you think this might not be enough? 🤔

Reply:
Yes - you are right, I had a check and the access is protected. In this case, I think doing this might remedy crashing the app, but it means that we are probably covering up another important bug - As far as I can see the only place `residualMembers` is null can be null is where it's declared and in `endSync` method. In this case it means either `endSync` is invoked early (before `startSync`) or is invoked after another `endSync` which would be problematic. I will approve this to remedy the crash but I think we should create another issue to investigate this situation.